app/src/main/java/id_iot/demo/CollectionFragment.java

```diff
@@ -0,0 +1,64 @@
+package id_iot.demo;
+
+import android.os.Bundle;
+import android.support.v4.app.Fragment;
+import android.support.v4.app.FragmentActivity;
+import android.support.v7.widget.SearchView;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.AdapterView;
+import android.widget.GridView;
+import android.widget.Toast;
+
+import java.util.ArrayList;
+
+public class CollectionFragment extends Fragment {
+
+    private GridView collectionView;
+    private ArrayList<Recipe> recipes;
+    SearchView findRecipe;
+    private CollectionAdapter collectionAdapter;
+
+    @Override
+    public View onCreateView(LayoutInflater inflater,
```

Comment:
overall this method is too long. Should be broken to smaller. like setupCollectionView or something like that